add nginx ML module tutorial (#3878)
elastic/observability-docs: docs/en/observability/monitor-nginx-ml.asciidoc

[discrete]
[[monitor-nginx-ml]]
== Part 5: Find anomalies in your nginx access logs

> The {integrations-docs}/nginx[Nginx Elastic integration] to help find unusual activity in your nginx access logs.

Reviewer: Missing a word here. Maybe "use"? Also you lowercased nginx Elastic integration above.

```suggestion
Use the {integrations-docs}/nginx[nginx Elastic integration] to help find unusual activity in your nginx access logs.
```

> Use the nginx machine learning (ML) module from the {integrations-docs}/nginx[Nginx Elastic integration] provides built in ML jobs to help find unusual activity in your nginx access logs.

Reviewer 2: Is all lowercase nginx the standard we're following now?

> Use the nginx machine learning (ML) module from the {integrations-docs}/nginx[Nginx Elastic integration] provides built in ML jobs to help find unusual activity in your nginx access logs.
> Monitoring anomalies in your access logs helps you detect:

Reviewer 2: ```suggestion
Use the nginx machine learning (ML) module from the {integrations-docs}/nginx[Nginx Elastic integration] to find unusual activity in your nginx access logs.
Monitoring anomalies in your access logs helps you detect:
```

* security threats
* network issues
* system performance issues
* operational efficiency issues

[discrete]
[[monitor-nginx-ml-jobs]]
=== nginx anomaly detection jobs

The nginx ML module provides the following jobs for detecting anomalies in your access logs:

[[horizontal]]

> Low request rates (low_request_rate_nginx):: Uses the {ml-docs}/ml-count-functions.html#ml-count[`low_count`] function to detect abnormally low request rates. Abnormally low request rates might indicate network issues or other issues are preventing requests from reaching the server.

Reviewer 2: ```suggestion
Low request rates (`low_request_rate_nginx`):: Uses the {ml-docs}/ml-count-functions.html#ml-count[`low_count`] function to detect abnormally low request rates. Abnormally low request rates might indicate that network issues or other issues are preventing requests from reaching the server.
```

> Unusual source IPs - high request rates (`source_ip_request_rate_nginx`):: Uses the {ml-docs}/ml-count-functions.html#ml-count[`hight_count`] function to detect abnormally high request rates from individual IP addresses. Many requests from a single IP or small group of IPs might indicate something malicious like a DDoS attack where many requests are sent to overwhelm the server and make it unavailable to users.

Reviewer 2: My brain would prefer "a large number of requests" over "many request" but I can't explain why. I guess because "many" somehow seems less precise? Feel free to ignore if you don't see what I mean.

Unusual source IPs - high distinct count of URLs (`source_ip_url_count_nginx`):: Uses the {ml-docs}/ml-count-functions.html#ml-distinct-count[`high_distinct_count`] function to detect individual IP addresses accessing abnormally high numbers of unique URLs. A single IP accessing many unique URLs might indicate something malicious like web scraping or an attempt to find sensitive data or vulnerabilities.

> Unusual status code rates (`status_code_rate_nginx`):: Uses the {ml-docs}/ml-count-functions.html#ml-count[`count`] function to detect abnormal status code rates. A high rate of status codes could indicate problems with broken links, bad URLs, or unauthorized access attempts. A high rate of status codes could also point to server issues like limited resources or bugs in your code.

Reviewer 3: I guess, by the phrase `to detect abnormal status code rates.` , you meant - abnormally high error or failure status code. It would be good if either word `error` or `failure` is added additionally.

Unusual visitor rates (`visitor_rate_nginx`):: Uses the {ml-docs}/ml-count-functions.html#ml-nonzero-count[`non_zero_count`] function to detect abnormal visitor rates. High visitor rates could indicate something malicious like a DDoS attack.
Low visitor rates could indicate issues with access to the server.

> Refer to {integrations-docs}/nginx#ml-modules[nginx integration ML modules] for more on the jobs and the ML module manifest.

Reviewer: This link has even less information on the jobs listed above. It does link to the ML module manifest. Can we remove the job part? Although, that begs a new question: What is the ML module manifest? This is the first time it's mentioned. So maybe even something like this?

```suggestion
NOTE: These anomaly detection jobs are available when you have data that matches the query specified in the ML module manifest. Users not following this tutorial can refer to {integrations-docs}/nginx#ml-modules[nginx integration ML modules] for more about the ML module manifest.
```

[discrete]
[[monitor-nginx-ml-prereqs]]
=== Before you begin

Verify that your environment is set up properly to use the {ml-features}.

> If the {es} {security-features} are enabled, you need a user with permissions to manage {anomaly-jobs}.

Reviewer 2: ```suggestion
If {es} {security-features} are enabled, you need a user with permissions to manage {anomaly-jobs}.
```
The article seems unnecessary here?

Refer to {ml-docs}/setup.html[Set up ML features].

[discrete]
[[monitor-nginx-ml-add-jobs]]
=== Add nginx ML jobs

Add the nginx ML jobs from the nginx integration to start using anomaly detection:

. From the main {kib} menu, go to *Machine Learning* → *Jobs*.
. Select *Create job*.
. In the search bar, enter *nginx* and select *Nginx access logs [Logs Nginx]*.
. Under *Use preconfigured jobs*, select the *Nginx access logs* card.
. Select *Create jobs*.

> Back on the *Jobs* page, you should see the nginx anomaly detection jobs—`low_request_rate_nginx`, `source_ip_request_rate_nginx`, s`ource_ip_url_count_nginx`, `status_code_rate_nginx`, and `visitor_rate_nginx`.

Reviewer 3: Please correct : s`ource_ip_url_count_nginx`,

> Back on the *Jobs* page, you should see the nginx anomaly detection jobs—`low_request_rate_nginx`, `source_ip_request_rate_nginx`, s`ource_ip_url_count_nginx`, `status_code_rate_nginx`, and `visitor_rate_nginx`.

Reviewer 4: @mdbirnstiehl This comment has been resolved but the typo is still in the text.

> Back on the *Jobs* page, you should see the nginx anomaly detection jobs—`low_request_rate_nginx`, `source_ip_request_rate_nginx`, s`ource_ip_url_count_nginx`, `status_code_rate_nginx`, and `visitor_rate_nginx`.

Author: Thanks, forgot to push my commit 😊
